### PR TITLE
Capture simulator output and forward to logger

### PR DESCRIPTION
**Pull request by themperek**
Not an ultimate solation but should be better than before.

### CODE REVIEW

**Comment by themperek** [comment]
@Losiek What you think? Can you test this?

File: cocotb_test/run.py [hunk 1]
```diff
@@ -40,8 +41,11 @@ def run(simulator=None, **kwargs):
 
     results_xml_file = sim.run()
 
+    sys.tracebacklimit = 0 # remove not needed traceback form assert
```
Comment (cmarqu) [inline cocotb_test/run.py]:
```suggestion
    sys.tracebacklimit = 0  # remove not needed traceback from assert
```

File: cocotb_test/simulator.py [hunk 2]
```diff
@@ -210,29 +215,32 @@ def get_abs_paths(self, paths):
...
-                    sys.stdout.write(out.decode("utf-8"))
-                    sys.stdout.flush()
+                
+                log_out = out.decode("utf-8").rstrip()
+                if log_out != "":
+                    self.logger.info(log_out)
 
-        return output_log
+            if process.returncode:
+                self.logger.error("Command termindated with error %d" % process.returncode)
```
Comment (cmarqu) [inline cocotb_test/simulator.py]:
```suggestion
                self.logger.error("Command terminated with error %d" % process.returncode)
```

File: cocotb_test/simulator.py [hunk 3]
```diff
@@ -354,7 +362,7 @@ def build_command(self):
                 cmd.append(["vsim"] + ["-c"] + ["-do"] + [do_script])
 
         else:
-            print("Skipping compilation:" + out_file)
+             self.logger.warning("Skipping compilation:" + out_file)
```
Comment (cmarqu) [inline cocotb_test/simulator.py]:
```suggestion
            self.logger.warning("Skipping compilation:" + out_file)
```

**Comment by eric-wieser** [comment]
Won't this result in nested loggers?

**Comment by themperek** [comment]
>Won't this result in nested loggers?

Yes. This was easy solution for now. Should find a better way. May require some switch inside cocotb? Any ideas ?

**Comment by eric-wieser** [comment]
Cocotb could direct its raw `LogRecord`s to a unix socket, and you could receive them

**Comment by themperek** [comment]
It would probably work but would be quite a workaround. 

Was thinking maybe to encode every logged line like `level:file_name:line_nuber:message" forward this to higher level and decode (custom longer manipulation)?

**Comment by themperek** [comment]
Or even send `LogRecords` directly to `pytest` somehow via `stdout`.

**Comment by eric-wieser** [comment]
Note that python has essentially built all the infrastructure for you to use a unix socket

**Comment by themperek** [comment]
OK. Something to investigate since never used it.

**Pull request merged**